haystack/schema.py

```diff
@@ -92,11 +92,12 @@ def __init__(
         self.content_type = content_type
         self.score = score
         self.meta = meta or {}
-        self.id_hash_keys = id_hash_keys
 
         allowed_hash_key_attributes = ["content", "content_type", "score", "meta", "embedding" ]
-        if not set(self.id_hash_keys) <= set(allowed_hash_key_attributes):
-            raise ValueError(f"You passed custom strings {id_hash_keys} to id_hash_keys which is deprecated. Supply instead a list of Document's attribute names that the id should be based on (e.g. {allowed_hash_key_attributes}). See https://github.com/deepset-ai/haystack/pull/1910 for details)")
+        self.id_hash_keys = id_hash_keys
+        if self.id_hash_keys != None:
+            if not set(self.id_hash_keys) <= set(allowed_hash_key_attributes):
```

Comment:
As this line is still failing with mypy I'd say we put a `# type:  ignore` here at the end of the line and maybe another comment explaining that mypy fails because it thinks set(None) might be called.

Reply:
👍  I added a comment at that line :)